packages/flutter/test/material/input_decorator_test.dart

```diff
@@ -1931,6 +1939,50 @@ void main() {
     expect(getBorder(tester), disabledBorder);
   });
 
+  testWidgets('OutlineInputBorder radius carries over when lerping', (WidgetTester tester) async {
+    const Key key = Key('textField');
+
+    await tester.pumpWidget(
+      const MaterialApp(
+        home: Material(
+          child: Directionality(
+            textDirection: TextDirection.ltr,
+            child: TextField(
+              key: key,
+              decoration: InputDecoration(
+                fillColor: Colors.white,
+                filled: true,
+                border: UnderlineInputBorder(
+                  borderSide: BorderSide(color: Colors.blue, width: 2.0),
+                  borderRadius: BorderRadius.zero,
+                ),
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    // TextField has the given border and doesn't ever tween to the default
+    expect(getBorderRadius(tester), BorderRadius.zero);
```

Comment:
It would be useful to verify that the border is initially zero, however there's no animation here.